maps_dashboards/public/model/layersFunctions.ts

```diff
@@ -80,7 +83,25 @@ export const getMaplibreBeforeLayerId = (
   }
   return undefined;
 };
+
+export const layerExistInMbSource = (layerConfigId: string, maplibreRef: MaplibreRef) => {
+  const layers = getCurrentStyleLayers(maplibreRef);
+  for (const layer in layers) {
+    if (layers[layer].id.includes(layerConfigId)) {
+      return true;
+    }
+  }
+  return false;
+};
+
 export const layersTypeIconMap: { [key: string]: string } = {
   [DASHBOARDS_MAPS_LAYER_TYPE.OPENSEARCH_MAP]: DASHBOARDS_MAPS_LAYER_ICON.OPENSEARCH_MAP,
   [DASHBOARDS_MAPS_LAYER_TYPE.DOCUMENTS]: DASHBOARDS_MAPS_LAYER_ICON.DOCUMENTS,
+  [DASHBOARDS_MAPS_LAYER_TYPE.CUSTOM_MAP]: DASHBOARDS_MAPS_LAYER_ICON.CUSTOM_MAP,
+};
+
+export const referenceLayerTypeLookup = {
+  [DASHBOARDS_MAPS_LAYER_TYPE.OPENSEARCH_MAP]: true,
+  [DASHBOARDS_MAPS_LAYER_TYPE.CUSTOM_MAP]: true,
+  [DASHBOARDS_MAPS_LAYER_TYPE.DOCUMENTS]: false,
```

Comment:
why not include this as property of Specification?

Reply:
> WMS from Coordinate maps looks like below, Shall we keep same fields/description? ![Screen Shot 2023-01-01 at 9 35 48 PM](https://user-images.githubusercontent.com/11067894/210197930-cfc1e24c-39a8-4c7d-9aeb-cec54f57cb8d.png)

Leaflet is using ` L.tileLayer.wms` to constructs the right WMS call from it's parameters and then forwards it to WMS server
```
var wmsLayer = L.tileLayer.wms('http://ows.mundialis.de/services/service?', {
    layers: 'TOPO-OSM-WMS'
}).addTo(map);
```

but in maplibre([example](https://maplibre.org/maplibre-gl-js-docs/example/wms/)), it doesn't have this kind of api call, we need explicitly pass parameters to it.

Reply:
if it is not possible to abstract those extra parameters similar to leaflet, will go with what we have. Shall we mark what is required and optional as well?

Reply:
Yes, just updated, also added the help text to each parameter